step-functions/nino_check.asl.json

```diff
@@ -339,7 +339,19 @@
           }
         }
       },
-      "Next": "Send Retry"
+      "Next": "Was this the users last attempt?",
+      "ResultPath": "$.saveFailedAttempt"
+    },
+    "Was this the users last attempt?": {
+      "Type": "Choice",
+      "Choices": [
+        {
+          "Variable": "$.check-attempts-exist.Count",
+          "NumericGreaterThanEquals": 1,
+          "Next": "Fetch Auth Code Expiry"
+        }
+      ],
+      "Default": "Send Retry"
```

Comment:
I see the ACs say 
```
THEN the NINO CRI generates an Auth Code
AND a 200 is returned to the Frontend to continue the OAuth flow
```

I'm wondering how the 422 returned from the "Send retry" step links to the auth code being generated and a 200 being returned? Feels like I'm missing something!

Reply:
It's hard to entirely visualise the flow by just looking at the code. I'll give you a run through on Monday.